src/main/java/com/datalogics/pdf/samples/extraction/TextExtract.java

```diff
@@ -82,28 +86,25 @@ public static void main(final String... args) throws Exception {
     public static void extractTextReadingOrder(final URL inputUrl, final URL outputUrl)
                     throws PDFInvalidDocumentException, PDFIOException, PDFFontException, PDFSecurityException,
                     UnsupportedEncodingException, IOException, PDFUnableToCompleteOperationException {
-        File outputFile = null;
-        try {
-            outputFile = new File(outputUrl.toURI());
-        } catch (final URISyntaxException e) {
-            throw new PDFIOException(e);
-        }
-
-        if (outputFile.exists()) {
-            Files.delete(outputFile.toPath());
-        }
-
         PDFDocument document = null;
-        try (final FileOutputStream outputStream = new FileOutputStream(outputFile)) {
+        try {
             document = DocumentUtils.openPdfDocument(inputUrl);
 
             final PDFFontSet docFontSet = FontUtils.getDocFontSet(document);
             final ReadingOrderTextExtractor extractor = ReadingOrderTextExtractor.newInstance(document, docFontSet);
             final WordsIterator wordsIter = extractor.getWordsIterator();
 
-            while (wordsIter.hasNext()) {
-                final Word word = wordsIter.next();
-                outputStream.write(word.toString().getBytes("UTF-8"));
+            if (wordsIter.hasNext()) {
+                try (final FileOutputStream outputStream = obtainOutputStream(outputUrl)) {
+                    do {
+                        final Word word = wordsIter.next();
+                        outputStream.write(word.toString().getBytes("UTF-8"));
+                    } while (wordsIter.hasNext());
+                }
+            } else {
+                if (LOGGER.isLoggable(Level.INFO)) {
+                    LOGGER.info(inputUrl.getFile() + " did not have any text to extract.");
```

Comment:
To avoid getting %-escapes in the filename, it's better to use ``inputUrl.toURI().getPath()`` here.